### PR TITLE
Pre-allocate arrays

### PR DESCRIPTION
**Pull request by chadhietala**

### CODE REVIEW

**Comment by jmeas** [comment]
Looks good!

File: lib/route-recognizer.js [hunk 1]
```diff
@@ -385,11 +386,14 @@ RouteRecognizer.prototype = {
...
   handlersFor: function(name) {
-    var route = this.names[name], result = [];
-    if (!route) { throw new Error("There is no route named " + name); }
+    var route = this.names[name];
 
+    if (!route) { throw new Error("There is no route named " + name); }
+    
+    var result = new Array(route.handlers.length);
+    
     for (var i=0, l=route.handlers.length; i<l; i++) {
```
Comment (stefanpenner) [inline lib/route-recognizer.js]: lets not cache length

**Comment by stefanpenner** [comment]
left one nit-pick, this looks good. It brings up a good point, it would be good to add a perf suite to this project at some point. I suspect lots of low hanging fruit here...

File: lib/route-recognizer.js (hunk 1 above)
Comment (chadhietala) [inline lib/route-recognizer.js]: Fixed all of these.

**Comment by chadhietala** [comment]
@stefanpenner I was going to try to setup a simple bench suite with https://github.com/stefanpenner/do-you-even-bench

**Comment by chadhietala** [comment]
Such checked in `dist`.

**Comment by chadhietala** [comment]
Decent bump in `add` ~100op/s.
```
  Add ................... 529.43 op/s
  End-to-end ............ 245.79 op/s
  Generate ........ 4,800,431.13 op/s
  Handlers For ... 37,624,535.59 op/s
  Recognize ......... 355,982.85 op/s
```

**Comment by krisselden** [comment]
baseline numbers: https://github.com/tildeio/route-recognizer/pull/76

**Pull request merged**